event-management-frontend/src/components/eventCreateOrEdit/locationSection/SearchBar.tsx

```diff
@@ -24,36 +24,75 @@ interface Props {
   searchMarker: LatLngExpression[];
   setsearchMarker: (searchMarker: LatLngExpression[]) => void;
 }
+
+function getUserDataWithPromise(props: string) {
+  const searchString =
+    'https://api.openrouteservice.org/geocode/autocomplete?api_key=' +
+    process.env.REACT_APP_OPENROUTESERVICE_KEY +
+    '&text=';
+  var xhr = new XMLHttpRequest();
+  xhr.responseType = 'json';
+  return new Promise(function (resolve, reject) {
+    xhr.onreadystatechange = function () {
+      if (xhr.readyState == 4) {
+        if (xhr.status >= 300) {
+          reject('Error, status code = ' + xhr.status);
+        } else {
+          resolve(xhr.response);
+        }
+      }
+    };
+    xhr.open('GET', searchString.concat(props), true);
+    xhr.send();
+  });
+}
+
 const SearchBar = (props: Props) => {
   const classesSearch = useStylesSearchBar();
   const [flag, setFlag] = useState(true);
   const [suggestions, setSuggestions] = useState<LocationType[]>([]);
+  const [locations, setLocations] = useState<LocationType[]>([]);
   const { t } = useTranslation();
 
   useEffect(() => {
+    const l: LocationType[] = [];
```

Comment:
You should be more specific with the naming of variables, maybe `locationTypes` here

Reply:
Changed to foundLocations